Yosemite/Yosemite/PointOfSale/POSItemProvider.swift

```diff
@@ -24,5 +24,13 @@ extension POSItem {
 }
 
 public protocol POSItemProvider {
-    func providePointOfSaleItems() async throws -> [POSItem]
+    func providePointOfSaleItems(pageNumber: Int) async throws -> [POSItem]
```

Comment:
Good catch! Interestingly it doesn't error out on my machine, it just keeps running the target forever in the _"Testing WooCommerceTests"_ step without ever actually failing, crashing, or otherwise. Do you see a different behaviour in your machine? For reference I'm using Xcode 15.4 🤔 

In any case I updated them on https://github.com/woocommerce/woocommerce-ios/pull/14187/commits/2ccb8c576d8e850aa284ec80b6944aaf87c52de5 and now all of them run and pass locally and in CI

Reply:
Just ran the tests before the updates today, the failing test cases on CI also crashed and failed locally in Xcode 16. Yea, it's interesting how the build still succeeded without compiler errors from the mock not implementing the protocol exactly. At least it was detected in unit tests 🤔

Reply:
>  the build still succeeded without compiler errors from the mock not implementing the protocol exactly

Yep, I was a bit dumfounded by that, Xcode haven't been able to catch some of these protocol conformances very good 🤔 Thanks for testing!